### DISCUSSION TITLE
Actions Running on Forks

### DISCUSSION THREAD

**Discussion by baubakg** [adobe/phased-testing#63] [General]
Hi @krmahadevan ,

I started this discussion regarding the sonar analysis.

The sonar analysis is here:
https://sonarcloud.io/project/overview?id=adobe_phased-testing

**Comment by krmahadevan** [comment]
We can recheck this report after my PR gets merged, because I have cleaned up a lot of those problems.

**Comment by krmahadevan** [comment]
@baubakg - Please take a look at this thread https://community.sonarsource.com/t/github-actions-public-repo-you-re-not-authorized-to-run-analysis/32311/6

It seems that forks are going to have a problem running sonar. So maybe we should just the sonar scans on the main branch and not on PRs. I do understand the intent behind having them, but when the tool lacks the support for the intent, it's better to remove the intent than have false failures for every PR.

here's a possible workaround which checks if the token is available and runs sonar scan ONLY if its available https://github.com/pf4j/pf4j/issues/457#issuecomment-861143092

**Reply by krmahadevan** [reply]
And this link https://stackoverflow.com/a/39720346 which explains/suggests that sonar scans should be run on the integration branches and not on PRs

**Comment by baubakg** [comment]
Ceated issue for this #72
